src/App.tsx

```diff
@@ -1,13 +1,29 @@
 import React from 'react';
+import {
+  Switch,
+  Route,
+} from "react-router-dom";
 import Home from './components/Home';
-
+import Settings from './components/Settings';
+import Header from './components/Header';
+import Footer from './components/Footer';
+import Login from './components/Login';
+import Article from './components/Article';
 
 
 function App() {
   return (
+    <main><Header />
     <div>
-      <Home />
-    </div>
+    <Switch>
```

Comment:
I make another comment about this further down.

I would strongly recommend that you put the Router with the Switch so that all the application routing is centralized in one place.

Reply:
Router is now colocated with Switch in this component